### ISSUE TITLE
Caldav event is not updated on reschedule

### ISSUE THREAD

**Issue by TheCapsLock** [calcom/cal.com#2988]
Found a bug? Please fill out the sections below. 👍

### Issue Summary

When any attendee (or organizer) reschedules an event, the update is not replicated on caldav side

### Steps to Reproduce

1. Create a new schedule (I used optin ones)
2. once booked, reschedule using the Web UI or link in the Email
3. Event is correctly updated in Cal.com's Web UI but  no on caldav side

### Technical details

- Firefox 100.0.2 ; cal.com v1.6.3 ; node v14.19.3
- Caldav server : nextcloud 24.0.1

**Comment by milospuac** [comment]
Hi there. Thanks for reaching out.  Our team will investigate.

**Comment by noobyogi0010** [comment]
Hey, @milospuac !
I would like to work on this issue.
Thanks!

**Comment by jdaviescoates** [comment]
> Also, in my case I ended up with _two_ booking in cal.com (but only the un-reschdeuled original event on my Nextcloud CalDAV calendar)

Just to add, I tried this a few more times and I no longer get an error, nor two events in my Cal.com bookings,  but I end up two events in my Nextcloud calendar: the original booking,  plus the rescheduled one.

**Comment by alishaz-polymath** [comment]
@jdaviescoates I think any changes (reschedule or location update) results in an extra updated booking in the nextcloud calendar which is definitely unintended. We've identified potential causes and currently working on getting this sorted. Thanks for your patience 🙏

**Comment by jdaviescoates** [comment]
@alishaz-polymath yeah, seems that way.

I have another similar issue too, but not sure if it warrants a new issue or not?

Bookings that are cancelled via the Cal.com web app are not deleted from my Nextcloud CalDAV calendar.

**Comment by alishaz-polymath** [comment]
Hey @jdaviescoates thanks for the update. I think there's a general fix we might merge in early this week which might end up fixing this ticket itself, or at least help us speed up the fix for this. 
As for the new issue you mentioned, please raise a new ticket and share 🙌

**Comment by jdaviescoates** [comment]
Just tried this again, rescheduling still doesn't work and I still get this error:

`Could not reschedule the meeting. JSON.parse: unexpected end of data at line 1 column 1 of the JSON data`

But at least no duplicate event was created!

**Comment by alishaz-polymath** [comment]
Mate it isn’t merged yet. Please check the PR status and test it once it is merged and deployed 🙂

**Comment by alishaz-polymath** [comment]
@jdaviescoatesr please book a slot with me [Here](https://cal.com/ali)  and we'll have a go at it together as I couldn't re create it 🙏

**Comment by alishaz-polymath** [comment]
Fix merged earlier, somehow didn't auto close

**Comment by jdaviescoates** [comment]
This is still exactly the same for me. Just tested again and got this:

`Could not reschedule the meeting. JSON.parse: unexpected end of data at line 1 column 1 of the JSON data`

**Comment by alishaz-polymath** [comment]
@jdaviescoates Accepted your call request. We'll have a go at it during our scheduled call 🙌

**Comment by jdaviescoates** [comment]
Thanks. I just thought perhaps my browser/ Cal was getting confused by me testing bookings with myself all in the same browser window so just tried again without doing that. Still got an error, but a different one:

``Could not reschedule the meeting. Unexpected end of JSON input``

**Comment by jdaviescoates** [comment]
> @jdaviescoates Accepted your call request. We'll have a go at it during our scheduled call raised_hands

I'm in the call, are you joining :) thanks!

**Comment by alishaz-polymath** [comment]
Hey @jdaviescoates, the fix is in review. You can keep track [here](https://github.com/calcom/cal.com/pull/5193) 
Thanks for providing access to speed this up :)

**Comment by jdaviescoates** [comment]
Great, thanks @alishaz-polymath, let us know when it's gone live on Cal.com so we can test it.

Presumably I can now delete your account on my system now? :)

**Comment by alishaz-polymath** [comment]
Hey @jdaviescoates 
It should be live on Monday/Tuesday. And I would keep the test account just in case. 🙏

**Comment by jdaviescoates** [comment]
> It should be live on Monday/Tuesday.

It's Wednesday and I'm still having the same issue... hopefully that's just because the merge isn't live on Cal.com yet?

**Comment by alishaz-polymath** [comment]
Hey @jdaviescoates it was merged yesterday so you definitely shouldn't be seeing that issue now. Could you please retest and maybe share a loom video? 🙏

**Comment by jdaviescoates** [comment]
Worked when I tested again so I think it's all fine! Many thanks for your work on this!

**Comment by alishaz-polymath** [comment]
@jdaviescoates Amazing 😃 
Thank you for the bug report. Please feel free to let me know if you face any other caldav related bugs at all 🙏
